packages/deployed-backend-client/API.md

```diff
@@ -126,7 +140,9 @@ export enum ConflictResolutionMode {
 
 // @public (undocumented)
 export type DeployedBackendClient = {
-    listBackends: (listBackendsRequest?: ListBackendsRequest) => Promise<ListBackendsResponse>;
+    listBackends: (listBackendsRequest?: ListBackendsRequest) => AsyncGenerator<{
+        backends: BackendSummariesMetadata[];
+    }, void, unknown>;

```

Comment:
while this will work, I think it would be a nicer API to consume if it returned `AsyncGenerator<BackendSummariesMetadata>`. This would allow iterating over the return value like
```ts
for await (const entry of listBackends(props)) {
  // each entry is a backend summary
}
```
instead of having to do a nested loop
```ts
for await (const responseChunk of listBackends(props)) {
  for (const entry of responseChunk.backends) {
    // get the individual entries here
  }
}
```

Reply:
I added my comments here https://github.com/aws-amplify/amplify-backend/pull/1096#discussion_r1517952728 .

TLDR. API should still allow caller handled pagination while offering easier way to craft for loop.
I think our goal should be to enable writing for loop without nesting, and that is DX we should optimize for. But there should be option to iterate page by page (which is secondary scenario and can be a bit more obscure).